src/Analyzers/CSharp/Analyzers/RemoveUnreachableCode/RemoveUnreachableCodeHelpers.cs

```diff
@@ -13,10 +14,23 @@ internal static class RemoveUnreachableCodeHelpers
     public static ImmutableArray<ImmutableArray<StatementSyntax>> GetSubsequentUnreachableSections(StatementSyntax firstUnreachableStatement)
     {
         ImmutableArray<StatementSyntax> siblingStatements;
+        BlockSyntax? unreachableStatementContainingBlock = null;
         switch (firstUnreachableStatement.Parent)
         {
             case BlockSyntax block:
-                siblingStatements = [.. block.Statements];
+                var topmostConsecutiveBlockParent = block;
+                while (true)
+                {
+                    var topmostParent = topmostConsecutiveBlockParent.Parent;
+                    if (!topmostParent.IsKind(SyntaxKind.Block))
+                    {
+                        break;
+                    }
+                    topmostConsecutiveBlockParent = (BlockSyntax)topmostParent;
+                }
```

Comment:
```suggestion
                while (topmostConsecutiveBlockParent.Parent is BlockSyntax parentBlock)
                    topmostConsecutiveBlockParent = parentBlock;
```

BUt also needs explanation of what's going on here.